compass/landice/mesh.py

```diff
@@ -1026,106 +1027,75 @@ def interp_ais_bedmachine(self, data_path, mali_scrip, nProcs, dest_file):
-                         'BedMachineAntarctica_2020-07-15_v02_edits_floodFill_extrap_fillVostok.nc'),  # noqa
-            '-d', dest_file,
-            '-m', 'e',
-            '-w', 'BedMachine_to_MPAS_weights.nc']
-    check_call(args, logger=logger)
+    def __guess_scrip_name(filename):
 
+        # try searching for string followed by a version number
+        match = re.search(r'(^.*[_-]v\d*[_-])+', filename)
 
-def interp_ais_measures(self, data_path, mali_scrip, nProcs, dest_file):
-    """
-    Interpolates MEASURES ice velocity dataset
-    to a MALI mesh
+        if match:
+            # slice string to end of match minus one to leave of final _ or -
+            base_fn = filename[:match.end() - 1]
+        else:
+            # no matches were found, just use the filename (minus extension)
+            base_fn = os.path.splitext(filename)[0]
 
-    Parameters
-    ----------
-    data_path : str
-        path to AIS datasets, including BedMachine
+        return f"{base_fn}.scrip.nc"
 
-    mali_scrip : str
-        name of scrip file corresponding to destination MALI mesh
+    logger = self.logger
 
-    nProcs : int
-        number of processors to use for generating remapping weights
+    source_scrip = __guess_scrip_name(os.path.basename(source_file))
+    weights_filename = "gridded_to_MPAS_weights.nc"
 
-    dest_file: str
-        MALI input file to which data should be remapped
-    """
+    if variables != "all":
+        # make sure this is a list
 
-    logger = self.logger
+        # if list, then join the list making it a space seprated list for cli
+        variables = " ".join(variables)
 
-    logger.info('creating scrip file for velocity dataset')
+    logger.info('creating scrip file for source dataset')
     # Note: writing scrip file to workdir
     args = ['create_SCRIP_file_from_planar_rectangular_grid.py',
-            '-i',
-            os.path.join(data_path,
-                         'antarctica_ice_velocity_450m_v2_edits_extrap.nc'),
-            '-s',
-            'antarctica_ice_velocity_450m_v2.scrip.nc',
-            '-p', 'ais-bedmap2',
+            '-i', source_file,
+            '-s', source_scrip,
+            '-p', proj,
             '-r', '2']
     check_call(args, logger=logger)
 
     # Generate remapping weights
+    # Testing shows 5 badger/grizzly nodes works well.
+    # 2 nodes is too few. I have not tested anything in between.
     logger.info('generating gridded dataset -> MPAS weights')
     args = ['srun', '-n', nProcs, 'ESMF_RegridWeightGen',
-            '--source',
-            'antarctica_ice_velocity_450m_v2.scrip.nc',
+            '--source', source_scrip,
             '--destination', mali_scrip,
-            '--weight', 'measures_to_MPAS_weights.nc',
+            '--weight', weights_filename,
             '--method', 'conserve',
             "--netcdf4",
             "--dst_regional", "--src_regional", '--ignore_unmapped']
     check_call(args, logger=logger)
```

Comment:
Trevor's call to `ESMF_RegridWeightGen` has the additional flag: `"-i", "-64bit_offset",`, which is missing here. Is that flag needed for Greenland, but not Antarctica? If so I can add an option to this function to toggle it. 

Or should that flag be added to the default arguments here?

Reply:
I'm not sure if it's needed. We can test it both way and remove it if it's not necessary.

Reply:
Okay so the `-i` flag is just an alias for the `-64bit_offset` flag, which is already included here. So I'm definitely going to drop that. 

From the `ESMF_RegridWeightGen` documentation: 
>    --64bit_offset    - an optional argument specifying that the weight file
                        will be created in the NetCDF 64-bit offset format
                        to allow variables larger than 2GB.  Note the 64-bit
                        offset format is not supported in the NetCDF version
                        earlier than 3.6.0.  An error message will be generated
                        if this flag is specified while the application is
                        linked with a NetCDF library earlier than 3.6.0.

But we have the `--netcdf4` flag included, so I'm not totally sure if the 2GB variables are still a constraint in netcdf4. I'll do some looking to figure that one out.

Reply:
From the testing below,  the weights file for Antartica mesh is pretty large: 
```bash
$ du -sh ./gridded_to_MPAS_weights.nc
# 16G     ./gridded_to_MPAS_weights.nc
```

So doing a pretty crude loop over the variables in that file with:
```python 
import xarray as xr 

with xr.open_dataset("gridded_to_MPAS_weights.nc") as ds:
    for var in ds.variables:
        print(f"{var:<20}: {ds[var].nbytes/1e+9:1.3f} GB")
# ...
# yc_a                : 1.239 GB
# yc_b                : 0.003 GB
# xc_a                : 1.239 GB
# xc_b                : 0.003 GB
# yv_a                : 4.956 GB
# xv_a                : 4.956 GB
# ...
```
Looks like yv_a and xv_a are greater than 2GB, and that was generated using `--netcdf4` without `--64bit_offset`, so I think we are OK.

Also, per @matthewhoffman suggestion; I've taken a look through the [`pyremap` source code](https://github.com/MPAS-Dev/pyremap/blob/1836cdaa9220d81ca7880580402297f1b50dcfbf/pyremap/remapper.py#L216-L221) and looks like only the `` flag is used there. 

So, with all this I'm going to leave the `--64bit_offset` flag off since it seems unnecessary.

Reply:
Thanks for catching that!